core/src/test/java/org/elasticsearch/index/mapper/DocumentParserTests.java

```diff
@@ -70,6 +70,33 @@ public void testFieldDisabled() throws Exception {
         assertNotNull(doc.rootDoc().getField(UidFieldMapper.NAME));
     }
 
+    public void testDotsWithExistingMapper() throws Exception {
+        DocumentMapperParser mapperParser = createIndex("test").mapperService().documentMapperParser();
+        String mapping = XContentFactory.jsonBuilder().startObject().startObject("type").startObject("properties")
+            .startObject("foo").startObject("properties")
+            .startObject("bar").startObject("properties")
+            .startObject("baz").field("type", "integer")
+            .endObject().endObject().endObject().endObject().endObject().endObject().endObject().endObject().string();
+        DocumentMapper mapper = mapperParser.parse("type", new CompressedXContent(mapping));
+
+        BytesReference bytes = XContentFactory.jsonBuilder()
+            .startObject()
+            .field("foo.bar.baz", 123)
+            .startObject("foo")
+            .field("bar.baz", 456)
+            .endObject()
+            .startObject("foo.bar")
+            .field("baz", 789)
+            .endObject()
+            .endObject().bytes();
+        ParsedDocument doc = mapper.parse("test", "type", "1", bytes);
```

Comment:
let's check that `doc.dynamicMappingsUpdate()` is null? (ie. there is no mapping update)